### PR TITLE
exclude slow tests from sharding calc for linux-bionic-py3.7-clang9-test

### PR DESCRIPTION
**Pull request by clee2000**
Fixes #ISSUE_NUMBER

Sharding for linux-bionic-py3.7-clang9 previously included slow test times in the calculation for how long a test takes, causing the sharding to be uneven:

| Duration      | Count | Name|
| ----------- | ----------- | ---|
| 11.2m      | 221       |linux-bionic-py3.7-clang9 / test (default, 1, 2, linux.2xlarge)|
| 1.1h   | 218        | linux-bionic-py3.7-clang9 / test (default, 2, 2, linux.2xlarge)|

Numbers taken from https://hud.pytorch.org/metrics from 04/10/2022 12:20 PM to 04/17/2022 12:20 PM.

The duration of these jobs on this PR are 39m and 38m.

### CODE REVIEW

**Review by seemethere — APPROVED** [review]

File: tools/stats/s3_stat_parser.py [hunk 1]
```diff
@@ -210,9 +224,10 @@ def get_previous_reports_for_branch(branch: str, ci_job_prefix: str = "") -> Lis
         logger.info(f'Grabbing reports from commit: {commit}')
         summaries = get_test_stats_summaries_for_job(sha=commit, job_prefix=ci_job_prefix)
         for job_name, summary in summaries.items():
-            reports.append(summary[0])
-            if len(summary) > 1:
-                logger.warning(f'WARNING: Multiple summary objects found for {commit}/{job_name}')
+            if job_name[:len(job_name) - len('-test')] == _get_stripped_CI_job():
```
Comment (janeyx99) [inline tools/stats/s3_stat_parser.py:227]: could you switch this to a more explicit "if job_name doesn't have slow" and add a comment for why we should exclude slow tests? -->Also wouldn't this mean the slow tests would have no stats for sharding?

**Review by janeyx99 — APPROVED** [review]
awesome!

**Pull request closed without merging**